docs/code/introduction/machine_code.md

```diff
@@ -0,0 +1,257 @@
+---
+title: "How to add machine-wide custom code"
+---
+
+
+# How to add machine-wide custom code
+
+MPF contains a “CustomCode” concept which lets you add custom code to your game.
+
+CustomCode classes are Python modules that run at the “root” of your game. You can use them to do anything you want.
+
+Note that MPF also has the ability to run custom mode code which is code that is associated with a certain game mode and is generally only active when the mode it’s in is active. So if you just want to write your own custom game logic, you’ll probably use mode code.
```

Comment:
I would move this note to the end of this section, so the entire Custom Code section is explained and then the difference between it and Mode Code.

Reply:
I was thinking back and forth, it came from the original docs. I have removed it here and added a statement in the introduction. I believe that is now the best way in this new format.